api/src/main/java/org/eclipse/microprofile/rest/client/annotation/ClientHeaderParam.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.rest.client.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Used to specify an HTTP parameter that should be sent with the outbound request.
+ * When this annotation is placed at the interface level of a REST client interface, the specified header will be sent on each request for all
+ * methods in the interface.
+ * When this annotation is placed on a method, the header will be sent only for that method. If the same HTTP header is specified in an annotation
+ * for both the type and the method, only the header value specified in the annotation on the method will be sent.
+ * <p>
+ * The value of the header to send can be specified explicitly by using the <code>value</code> attribute.
+ * The value can also be computed via a default method on the client interface that returns a String value.  This method must be specified in the
```

Comment:
Are we also supporting {MyClass.myMethod} usage, or only default methods?

Reply:
My thought was only default methods.  That still gives users flexibility with extending other interfaces, etc.  Do you think we need to support {MyClass.myMethod} semantics?

Reply:
I think in the call we discussed utility methods too, so that you could have a common method to retrieve an authentication token, for instance, without having to replicate that code across all REST Client interfaces.

You could have the default method in multiple interfaces call that utility method, but that just seems like unnecessary boilerplate

Reply:
That makes sense.  So then the requirement for the compute method would be that it must either be a default method on the client interface, or a static method that is visible to the interface (i.e. not private, or package-protected in another package, etc.).

Reply:
Sounds good